src/main/java/pl/allegro/tech/embeddedelasticsearch/InstanceDescription.java

```diff
@@ -1,19 +1,30 @@
 package pl.allegro.tech.embeddedelasticsearch;
 
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Map;
+
 class InstanceDescription {
-    private final int port;
-    private final String clusterName;
+    private Map<String, Object> settings;
 
-    InstanceDescription(int port, String clusterName) {
-        this.port = port;
-        this.clusterName = clusterName;
+    InstanceDescription(Map<String, Object> settings) {
+        this.settings = settings;
     }
 
     int getPort() {
-        return port;
+        return (int) this.settings.get("transport.tcp.port");
     }
 
     String getClusterName() {
-        return clusterName;
+        return (String) this.settings.get("cluster.name");
```

Comment:
I will try, but I have some problem because sometimes in the code I see "this" and sometimes not :)